src/Hpack/Render/Dsl.hs

```diff
@@ -53,10 +74,11 @@ data RenderSettings = RenderSettings {
   renderSettingsIndentation :: Int
 , renderSettingsFieldAlignment :: Alignment
 , renderSettingsCommaStyle :: CommaStyle
+, renderSettingsEmptyLinesAsDot :: Bool
```

Comment:
I noticed the use of `Bool` here, rather than some sum datatype. In other circumstances, the code base has a number of sum datatypes that are isomorphic to `Bool`.

Reply:
Personally I am OK with  using a `Bool` value for a record field, as the field name describes the semantic.

But at the point you pass things around as positional arguments it's probably time to reach for a custom data time.

Her specifically, `RenderSettings` is exposed through the API, but any user specified `renderSettingsEmptyLinesAsDot` is always overruled.  So I think this should actually not be part of render settings, so maybe I should turn it into a positional argument and indeed use a custom data type.

Reply:
I ended up with https://github.com/sol/hpack/pull/626/commits/f8e64c55bad8caf954e50108b66293092dc26c2f